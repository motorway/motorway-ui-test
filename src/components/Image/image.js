import React from 'react';
import './image.scss';

const Image = (props) => {
  const { index, id, alt_description, url, user, likes, onClick } = props;
  return (
    <div onClick={() => onClick(id)} tabIndex={index} className="picture">
      <div className="img-wrapper">
        <img className="image" src={url} alt={alt_description} />
      </div>
      <div className="info">
        <div className="user">
          <img className="avatar" src={user.profile_image} alt="" />
          <p className="username">{user.username}</p>
        </div>
        {likes ? <p>❤️ {likes}</p> : <div />}
      </div>
    </div>
  );
};

export default Image;
