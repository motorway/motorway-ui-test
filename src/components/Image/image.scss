@import './../../styles/variables.scss';

.picture {
  position: relative;
  margin: 5px;
}

.img-wrapper {
  width: 300px;

  img {
    border-radius: 20px;
    cursor: pointer;
    display: block;
    height: 100%;
    width: 100%;
  }
}

.info {
  position: absolute;
  bottom: 0;
  color: var(--white-color);
  left: 0;
  right: 0;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1em;
  z-index: 1;
}

.user {
  display: flex;
  align-items: center;
  justify-content: center;
}

.avatar {
  border: 1px solid var(--white-color);
  border-radius: 50%;
  display: block;
  height: var(--avatar-size);
  margin-right: 1em;
  max-height: 300px;
  width: var(--avatar-size);
}
