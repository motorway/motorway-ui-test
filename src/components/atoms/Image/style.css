.image_container {
  display: flex;
  width: calc(calc(100% - 4rem) / 3);
  height: 20rem;
  margin-bottom: 2rem;
}

.image {
  width: 50%;
  height: 100%;
}
