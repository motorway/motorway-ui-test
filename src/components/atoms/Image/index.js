import React from "react";
import { Box } from "@material-ui/core";
import "./style.css";

const Image = ({ url, image }) => (
  <Box className="image_container">
    <img src={url} alt="" className="image" />
    <img src={image} alt="" className="image" />
  </Box>
);

export default Image;
