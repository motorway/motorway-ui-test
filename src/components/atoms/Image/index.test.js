import React from "react";
import { render, screen } from "@testing-library/react";
import Image from ".";

describe("Atoms: Images", () => {
  let props;
  let component;
  beforeEach(() => {
    props = { url: "fakeUrl", image: "fakeImageUrl" };
    component = render(<Image {...props} />);
  });
  it("Should match snapshot", () => {
    expect(component.asFragment()).toMatchSnapshot();
  });
  it("Should contain required amount of image elements", async () => {
    const images = await screen.findAllByRole("img");
    const got = images.length;
    const expected = 2;
    expect(got).toEqual(expected);
  });
});
