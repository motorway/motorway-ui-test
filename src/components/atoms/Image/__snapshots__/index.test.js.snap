// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms: Images Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-1 image_container"
  >
    <img
      alt=""
      class="image"
      src="fakeUrl"
    />
    <img
      alt=""
      class="image"
      src="fakeImageUrl"
    />
  </div>
</DocumentFragment>
`;
